modules/reindex/src/main/java/org/elasticsearch/index/reindex/remote/RemoteScrollableHitSource.java

```diff
@@ -221,15 +222,21 @@ protected void doRun() throws Exception {
                 }
 
                 @Override
-                public void onFailure(Exception t) {
-                    if (t instanceof ResponseException) {
-                        ResponseException re = (ResponseException) t;
+                public void onFailure(Exception e) {
+                    if (e instanceof ResponseException) {
+                        ResponseException re = (ResponseException) e;
                         if (RestStatus.TOO_MANY_REQUESTS.getStatus() == re.getResponse().getStatusLine().getStatusCode()) {
-                            listener.onRetryableFailure(t);
+                            listener.onRetryableFailure(e);
                             return;
                         }
+                        if (RestStatus.UNAUTHORIZED.getStatus() == re.getResponse().getStatusLine().getStatusCode()) {
```

Comment:
> I agree that my original idea of 4xx -> 400 is too simplistic.

I think that is on #17548, a different, but related place.

I could add a new exception that supports setting status. It might be as simple as renaming `ElasticsearchSecurityException`.